lib/monitoring/monitoringClientInterface.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const assert = require('assert')
+const client = require('prom-client');
+
+/**
+ * Wrapper for Prometheus client - https://github.com/siimon/prom-client
+ */
+
+var references = {};
+var namespace = '';
+
+module.exports = {
+  /**
+   * @param {string} ns - Sets a namespace for all the metrics
+   */
+  setNamespace: function(ns) {
+      assert(typeof ns === 'string');
+      namespace = ns;
+  },
+
+  /** getMetrics
+   *  Gets all the metrics in the form of a string for prometheus server
+   */
+  getMetrics: function() {
+    return client.register.metrics();
+  },
+
+  /** 
+   * Get metrics by its name
+   * @param {string} metricsName - name of the metrics
+   */
+  get: function (metricsName) {
+    return references[metricsName];
+  },
+  
+  /**
+   * Counters go up, and reset when the process restarts.
+   * @param {string} metricsName - name of the metrics
+   * @param {string} metricsHelp - description of metrics
+   * @param {string} metricsLabels - Static labels applied to every metric.
+   * Optional Parameter - metricsLabels 
+   */
+  createCounter: function (metricsName, metricsHelp, metricsLabels) {
+    if (metricsLabels === undefined) {
+        metricsLabels = [];
+    }
+    references[metricsName] = new client.Counter({
+        name: metricsName,
+        help: metricsHelp,
+        labels: metricsLabels
+    });
+    return references[metricsName];
+  },
+
+  /**
+   * Gauges are similar to Counters but Gauges value can be decreased.
+   * @param {string} metricsName - name of the metrics
+   * @param {string} metricsHelp - description of metrics
+   * @param {string} metricsLabels - Static labels applied to every metric
+   * Optional Parameter - metricsLabels 
+   */
+  createGauge: function (metricsName, metricsHelp, metricsLabels) {
+    if (metricsLabels === undefined) {
+        metricsLabels = [];
+    }
+    references[metricsName] = new client.Gauge({
+        name: namespace + '_' + metricsName, 
+        help: metricsHelp, 
+        labels: metricsLabels
+    });
+    return references[metricsName];
+  },
+  /**
+   * Histograms track sizes and frequency of events.
+   * The defaults buckets are intended to cover usual web/rpc requests,
+   * this can however be overriden.
+   * @param {string} metricsName - name of the metrics
+   * @param {string} metricsHelp - description of metrics
+   * @param {string} metricsBuckets - size of buckets for cumulative counters
+   * @param {string} metricsLabels - Static labels applied to every metric
+   * Optional Parameter - metricsBuckets and metricsLabels
+   */
+  createHistogram: function (metricsName, metricsHelp, metricsBuckets, metricsLabels) {
+    if (metricsBuckets === undefined) {
+        metricsBuckets = [0.1, 5, 15, 50, 100, 500]
```

Comment:
These don't match the defaults used by `prom-client`. Why these values instead?

Reply:
Changed to the default values: [0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10]